app/templates/app/styleguide.html

```diff
@@ -0,0 +1,35 @@
+---
+layout: style_guide
+title: "Style Guide"
+excerpt: "A handy collection of all the colors, typography, UI patterns, and components used on Made Mistakes."
+body_class: 'styleguide'
+---
+
+{% assign entries = site.colors %}
+{% assign componentsByType = site.components | group_by:"type" %}
+
+<nav id="component-selector" class="wrap">
+  <form>
+    <select name="newurl" id="component-select" onChange="window.location.replace(this.options[this.selectedIndex].value)">
+      <option value="">Select a Component</option>
+      <option value="#guide-color-palettes">Colors</option>
+      {% for type in componentsByType %}
+      <option value="#guide-{{ type.name }}">{{ type.name | capitalize }}</option>
+      {% for entry in type.items %}
+      <option value="#guide-{{ entry.title | slugify }}">&nbsp;&nbsp;&nbsp;{{ entry.title }}</option>
+      {% endfor %}
+      {% endfor %}
+    </select>
+  </form>
+</nav>
+
+<h2 id="guide-color-palettes" class="cf">Colors</h2>
+{% for entry in entries %}
+  {% include component-color.html %}
+{% endfor %}
+{% for type in componentsByType %}
+<h2 id="guide-{{ type.name }}" class="cf">{{ type.name | capitalize }}</h2>
+{% for entry in type.items %}
+{% include component.html %}
+{% endfor %}
+{% endfor %}
```

Comment:
My recommendation would be to indent each nested line. My personal approach is to write source code that is easy to read, not the compiled output. For example:

```liquid
{% for type in componentsByType %}
  <h2 id="guide-{{ type.name }}" class="cf">{{ type.name | capitalize }}</h2>
  {% for entry in type.items %}
    {% include component.html %}
  {% endfor %}
{% endfor %}
```